master/buildbot/www/oauth2.py

```diff
@@ -197,6 +196,19 @@ class GitHubAuth(OAuth2Auth):
     tokenUri = 'https://github.com/login/oauth/access_token'
     resourceEndpoint = 'https://api.github.com'
 
+    def __init__(self,
+                 clientId, clientSecret, serverURL=None, autologin=False, **kwargs):
+
+        OAuth2Auth.__init__(self, clientId, clientSecret, autologin, **kwargs)
+        if serverURL is not None:
+            # setup for enterprise github
+            if serverURL.endswith("/"):
+                serverURL = serverURL[:-1]
+
+            GitHubAuth.authUri = '{0}/login/oauth/authorize'.format(serverURL)
+            GitHubAuth.tokenUri = '{0}/login/oauth/access_token'.format(serverURL)
+            GitHubAuth.resourceEndpoint = '{0}/api/v3'.format(serverURL)
```

Comment:
Still need to change to self.xxx (dont change the class variables, change the instance variables)
then please add a unit test to keep the test coverage high.
https://docs.codecov.io/docs/browser-extension